draft-irtf-cfrg-hybrid-kems.md

```diff
@@ -273,20 +307,98 @@ of testing):
 We assume that the values produced and consumed by the above functions are all
 byte strings, with fixed lengths:
 
-- `Nseed`: The length of a key seed (input to DeriveKeyPair)
-- `Nek`: The length of a public encapsulation key
-- `Ndk`: The length of a secret decapsulation key
-- `Nct`: The length of a ciphertext produced by Encaps
-- `Nss`: The length of a shared secret produced by Encaps or Decaps
+- `Nseed`: The length in bytes of a key seed (input to DeriveKeyPair)
+- `Nek`: The length in bytes of a public encapsulation key
+- `Ndk`: The length in bytes of a secret decapsulation key
+- `Nct`: The length in bytes of a ciphertext produced by Encaps
+- `Nss`: The length in bytes of a shared secret produced by Encaps or Decaps
+
+## Nominal Groups {#groups}
+
+~~~ aasvg
+                            g
+                            |
+             +--------------+---------------+
+             |                              |
+             V                              V
+          +-----+                        +-----+
+       +->| Exp |                        | Exp |<-+
+       |  +-----+                        +-----+  |
+       |     |                              |     |
+       |     |                              |     |
+       |     V                              V     |
+       |    pkA                            pkB    |
+       |     |                              |     |
+ skA --+     +-------------.  .-------------+     +-- skB
+       |                    \/                    |
+       |                    /\                    |
+       |     +-------------'  '-------------+     |
+       |     |                              |     |
+       |     V                              V     |
+       |  +-----+                        +-----+  |
+       +->| Exp |                        | Exp |<-+
+          +-----+                        +-----+
+             |                              |
+             |                              |
+             V                              V
+           pkAB ========================= pkBA
+~~~
+
+Nominal groups are an abstraction that represents the core mechanics of
+Diffie-Hellman key exchange, including both finite field and elliptic-curve
+Diffie-Hellman {{ABH+21}}.  A nominal group comprises a set `G` together with a
+distinguished basis element `g`, an "exponentiation" map, and some auxiliary
+functions:
```

Comment:
This is incorrect, nominal groups do not align with finite fields especially regarding Diffie-Hellman - finite fields have other algebraic structure that ofc make them weaker to instantiate DH over than groups.

```suggestion
Nominal groups are an abstract model of elliptic curve groups, over which we
instantiate Diffie-Hellman key agreement {{ABH+21}}.  A nominal group comprises
a set `G` together with a distinguished basis element `g`, an "exponentiation"
map, and some auxiliary functions:
```